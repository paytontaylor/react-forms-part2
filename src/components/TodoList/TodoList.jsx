import { useState } from "react";
import NewTodoForm from "../NewTodoForm/NewTodoForm";
import Todo from "../Todo/Todo";

const TodoList = () => {
  const [todos, setTodos] = useState([]);

  const addTodo = (newTodo) => {
    setTodos(todos => ([...todos, newTodo]))
  }

  const deleteTodo = (todo) => {
    const updatedTodos = todos.filter(t => t !== todo);
    setTodos(updatedTodos);
  }

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>
      { todos.map(todo => <Todo key={todo} todo={todo} deleteTodo={deleteTodo} />) }
      </ul>
    </div>
  )
}

export default TodoList;