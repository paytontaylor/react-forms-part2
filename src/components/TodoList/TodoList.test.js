import { render, screen } from "@testing-library/react";
import TodoList from "./TodoList";

it("should render the TodoList without crashing", () => {
  render(<TodoList />)
})

it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
})

it("should display a list of todos", () => {
  render(<TodoList />)
})