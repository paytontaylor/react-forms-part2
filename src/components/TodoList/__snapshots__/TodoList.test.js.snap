// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="newTodo"
        >
          New Todo
        </label>
        <input
          id="newTodo"
          name="newTodo"
          type="text"
          value=""
        />
        <button>
          Add
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
